dissect/target/plugins/os/unix/log/journal.py

```diff
@@ -377,8 +381,8 @@ def journal(self) -> Iterator[JournalRecord]:
                         kernel_device=entry.get("kernel_device"),
                         kernel_subsystem=entry.get("kernel_subsystem"),
                         udev_sysname=entry.get("udev_sysname"),
-                        udev_devnode=path.from_posix(entry.get("udev_devnode")) if entry.get("udev_devnode") else None,
-                        udev_devlink=path.from_posix(entry.get("udev_devlink")) if entry.get("udev_devlink") else None,
+                        udev_devnode=get_optional(entry.get("udev_devnode"), path.from_posix),
+                        udev_devlink=get_optional(entry.get("udev_devlink"), path.from_posix),
                         journal_hostname=entry.get("hostname"),
                         filepath=path.from_posix(f.path),
```

Comment:
```suggestion
                        filepath=f.path,
```
This is now allowed with the latest stable `flow.record`

Reply:
Is this only not allowed for the field `filepath` or also for `udev_devnode` and `udev_devlink`. If so could a `DeprecationWarning` be introduced if this function is used?

Reply:
Any `path` flow.record type can now also correctly be instantiated directly from a compatible `Path`-like object (of which `TargetPath` is one). The `from_posix`/`from_windows` functions are still valid when needing to instantiate from e.g. a string.